classes/patterns/class-sideimagewithtextandcta.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Side image with text and CTA pattern class.
+ *
+ * @package P4GBKS
+ * @since 0.1
+ */
+
+namespace P4GBKS\Patterns;
+
+/**
+ * Class Side image with text and CTA.
+ *
+ * @package P4GBKS\Patterns
+ */
+class SideImageWithTextAndCta {
+
+	/**
+	 * Pattern name.
+	 *
+	 * @const string PATTERN_NAME.
+	 */
+	const PATTERN_NAME = 'side-image-with-text-and-cta';
+
+	/**
+	 * Side image with text and CTA pattern constructor.
+	 */
+	public function __construct() {
+		if ( ! function_exists( 'register_block_pattern' ) ) {
+			return;
+		}
+
+		register_block_pattern(
+			self::PATTERN_NAME,
```

Comment:
Ideally we have this logic somewhere else, so that we don't need to add more than just the config for every pattern.

WP core does this by including files that return an array.

https://github.com/WordPress/WordPress/blob/8475350f0e408d00832caedcfba340bed96f6266/wp-includes/block-patterns.php#L21-L37

Reply:
You could add an abstract class with `get_name`, and `get_config` abstract methods? It can include shared logic, and also allows IDEs to auto generate most of the class (at least in PHPStorm, apparently [you need to pay for a premium plugin](https://stackoverflow.com/questions/54883976/vs-code-auto-generate-interface-methods-in-php-class) to have this feature :cry: )

Like we use for [migration scripts](https://github.com/greenpeace/planet4-master-theme/blob/71347e57b872b5c27aaf8cc0c946790cd455f37a/src/MigrationScript.php#L13-L30).

Reply:
We can also do this once we add the next pattern.

Reply:
[I've added a base class for the patterns](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/789/commits/befa20906c7efde0765ddb1dae7edd2c9c5e78f1), similar to what we do for blocks... Anything else you think we should put there? 🤔